Doc update: how to autoload apps in flash

beardeddude opened the pull request:
I'm not sure if this is even good practice. But I learned I could do this by reading the software.py file. I tried to get it to work using the upload command of the wasp tool but couldn't get it to work. But by using the push command I am easily able to store a .mpy so that its automatically added to the software list. 

I personally found this useful but it obviously still has the RAM constraints of a flash based program. I didn't think the docs made it clear this was possible. So I offer this update. If this is encouraging bad practice I'm interested in that as well.

Signed-off-by: Adam Blair adampblair@protonmail.com

fgaz commented:
Thanks! This piece of documentation was really needed.

> I'm not sure if this is even good practice.

Yes it is. All apps in `apps/` are made to be uploaded that way.

>  I tried to get it to work using the upload command of the wasp tool but couldn't get it to work.

What was the error? You should use `--binary` because `.mpy` files are not just text.

You should also mention that that the app should follow a naming convention: the file name should be the class name - "App" + ".mpy"/".py", so "ReadMeApp" → "ReadMe.mpy"/"ReadMe.py"

For reference the feature was added in ba8546dd605799c189ccce778b0c468cb129054d

beardeddude commented:
Thanks for the feedback. I updated the change to use the binary upload (which was what I was missing). And added a section on the naming conventions as I had some trouble figuring them out myself. I think I got them right but sanity check me.
